src/pods/toolbar/components/export-button/export-button.business.ts

```diff
@@ -28,6 +28,78 @@ export const calculateTableEndYPosition = (table: TableVm): number => {
   return table.y + tableHeight;
 };
 
+export const isTableOverlap = (table: TableVm, tables: TableVm[]): boolean => {
+  if (tables.length === 0) {
+    return false;
+  }
+
+  const expandedFields = expandTableFields(table.fields);
+  const tableEndXPosition = table.x + TABLE_CONST.TABLE_WIDTH;
+  const tableEndYPosition = calculateTableEndYPosition({
+    ...table,
+    fields: expandedFields,
+  });
+
+  return tables.some(t => {
+    if (t.id === table.id) {
+      return false;
+    }
+
+    const tExpandedFields = expandTableFields(t.fields);
+    const tEndXPosition = t.x + TABLE_CONST.TABLE_WIDTH;
+    const tEndYPosition = calculateTableEndYPosition({
+      ...t,
+      fields: tExpandedFields,
+    });
+
+    const overlapInX = table.x <= tEndXPosition && tableEndXPosition >= t.x;
+    const overlapInY = table.y <= tEndYPosition && tableEndYPosition >= t.y;
+
+    return overlapInX && overlapInY;
+  });
+};
+
+export const placeTableWithoutOverlap = (
+  table: TableVm,
+  tables: TableVm[]
+): TableVm => {
+  let newTable = table;
+  let attempts = 0;
+
+  while (
+    isTableOverlap(newTable, tables) &&
+    attempts < TABLE_CONST.MAX_PLACEMENT_ATTEMPTS
+  ) {
+    newTable = {
+      ...newTable,
+      x: newTable.x - TABLE_CONST.ROW_HEIGHT,
```

Comment:
Why in the X coordinate we are substracting the ROW_HEIGHT (Y coordinate?

Reply:
I have created a new variable for this, TABLE_SHIFT_DISTANCE